components/style/servo/mod.rs

```diff
@@ -9,3 +9,15 @@
 pub mod media_queries;
 pub mod restyle_damage;
 pub mod selector_parser;
+
+use shared_lock::SharedRwLock;
+
+lazy_static! {
+    /// Per-process shared lock for author-origin stylesheets
+    ///
+    /// FIXME(SimonSapin): I wanted this to be per-document (or per-pipeline?)
+    /// but couldn’t figure out a way to get references to the same lock in both
+    /// the Document node and the LayoutThread.
+    /// Giving up to unblock.
```

Comment:
I believe the layout thread can access the doc at a given point. I think it should really be per-origin, but you need to somehow propagate the state from the pipeline to script. Probably it's not too hard, but could you file and add the issue number here?